### PR TITLE
Support to parse multiple jtl files

### LINKED ISSUE #808: Need to process multiple jtl files if exists in a test scenario
**Description:**

Some test scenarios generate multiple jtl files after tests have run. The existing parsing mechanism only supports a single jtl file in a scenario. 

**Affected Product Version:**
0.9.0-M31

**Related Issues:**

#806

### PR DESCRIPTION
**Pull request by harshanL**
**Purpose**

To resolve #808 

**Security checks**
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: automation/src/main/java/org/wso2/testgrid/automation/parser/FunctionalTestResultParser.java [hunk 1]
```diff
@@ -81,61 +82,69 @@ public FunctionalTestResultParser(TestScenario testScenario, String testLocation
...
-                        String nodeName = endElement.getName().getLocalPart();
-                        if (HTTP_SAMPLE_ELEMENT.equalsIgnoreCase(nodeName) ||
-                                SAMPLE_ELEMENT.equalsIgnoreCase(nodeName)) {
-                            this.testScenario.addTestCase(testCase);
-                        }
-                        break;
-                    default:
-                        break;
+        if (scenarioResultFiles.length == 0) {
+            logger.warn("Unable to locate jtl files for the scenario : " + testScenarioName);
```
Comment (kasunbg) [inline automation/src/main/java/org/wso2/testgrid/automation/parser/FunctionalTestResultParser.java]: Shall we print where you searched for files as well?

File: automation/src/main/java/org/wso2/testgrid/automation/parser/FunctionalTestResultParser.java [hunk 2]
```diff
@@ -81,61 +82,69 @@ public FunctionalTestResultParser(TestScenario testScenario, String testLocation
...
+            return;
+        }
+
+        for (String jtlFile : scenarioResultFiles) {
+            try (InputStream inputStream = new FileInputStream(Paths.get(testLocation, jtlFile).toString())) {
```
Comment (kasunbg) [inline automation/src/main/java/org/wso2/testgrid/automation/parser/FunctionalTestResultParser.java]: This is the API to get an InputStream in Java 7+ - `jav.nio.file.Files#newInputStream`

https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#newInputStream-java.nio.file.Path-java.nio.file.OpenOption...-

File: automation/src/main/java/org/wso2/testgrid/automation/parser/FunctionalTestResultParser.java [hunk 3]
```diff
@@ -81,61 +82,69 @@ public FunctionalTestResultParser(TestScenario testScenario, String testLocation
...
+                            if (HTTP_SAMPLE_ELEMENT.equalsIgnoreCase(elementName) ||
+                                    SAMPLE_ELEMENT.equalsIgnoreCase(elementName)) {
+                                testCase = this.buildTestCase(startElement);
+                            } else if (FAILURE_MESSAGE_ELEMENT.equalsIgnoreCase(elementName)) {
+                                failureMsgElement = true;
+                            }
+                            break;
+                        case XMLStreamConstants.CHARACTERS:
+                            Characters characters = event.asCharacters();
+                            if (failureMsgElement) {
```
Comment (kasunbg) [inline automation/src/main/java/org/wso2/testgrid/automation/parser/FunctionalTestResultParser.java]: What will happen if there was no characters inside FAILURE_MESSAGE_ELEMENT?

Comment (harshanL) [reply automation/src/main/java/org/wso2/testgrid/automation/parser/FunctionalTestResultParser.java]: I think JMeter always add an error message if there's a test case failure.

**Pull request merged**